app.py

```diff
@@ -105,28 +105,32 @@ def samples():
     #only return public data samples if not logged in
     filters = dict(request.args)
     tmp_var = 0
+    print("Pre-processing filters")
+    print(filters)
 
     for key in filters.keys():
+        # Key is a map polygon
         if key == "polygon_coords" and filters[key][0]:
-            coords = filters[key][0][1:-1].strip().split("],[")
+            # List of coordinate point strings; remove trailing comma, starting/trailing bracket,
+            #   and split into coordinate points
+            coords = filters[key][0].strip(',').strip('[').strip(']').split('],[')
```

Comment:
Is this kind of stripping  done in other places in the code? Perhaps it should be abstracted into a function for use later on